addons/hr_holidays/wizard/hr_departure_wizard.py

```diff
@@ -18,3 +19,8 @@ def action_register_departure(self):
                                                          ('date_to', '>', self.departure_date),
                                                          ('state', 'not in', ['cancel', 'refuse'])])
             future_leaves.write({'state': 'cancel'})
+
+        if self.archive_allocation:
+            employee_allocations = self.env['hr.leave.allocation'].search([('employee_id', '=', self.employee_id.id)])
+            if len(employee_allocations) > 0:
```

Comment:
You can remove the condition and just call action_archive() - it will do nothing if the recordset is empty